### PR TITLE
CI: Running tests with xvfb for the clipboard to work

### LINKED ISSUE #26887: CI: read_clipboard / to_clipboard not working in Azure
The clipboard system does not work in Azure Pipelines. I tried the next things:

Installing `xsel` and `xclip` (separately) as suggested in the documentation

Setting up a step in the azure settings based on the travis docs:
```
  - script: |
      set -x
      sudo apt-get install -y xsel xvfb
      export DISPLAY=":99.0"
      sh -e /etc/init.d/xvfb start
      sleep 3
    displayName: 'Setup clipboard'
```

But we still get the same paperclyp error. This is affecting the docs (see #26852) and also the tests (see #26428).

### PR DESCRIPTION
**Pull request by datapythonista**
- [X] closes #26887
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

With some luck this should make the clipboard tests work in azure. @jorisvandenbossche @jreback can you have a look once the CI finishes?

### CODE REVIEW

**Comment by datapythonista** [comment]
No idea why `xvfb` manages to start in some Linux builds, and fails in others. The only difference should be the conda packages installed, that I don't think it should affect `xvfb`.

And it also worries me, that in my fork pipelines (I set it up to test changes to the CI), things fail that don't fail here. I feel like I was dealing with windows 95... :\

**Comment by datapythonista** [comment]
In the build `Linux py37_locale` this seems to be working as expected for what I see. Clipboard tests are usually shown in the print skipped section, and here they are not. So looks like the problem is that there is something different in the `Linux py35_compat` build, that is making `xvfb` fail to start.

**Comment by datapythonista** [comment]
I see that `xvfb` actually starts the first time, and fails the second execution. I guess the problem may be that it doesn't have time to switch off before it's reloaded, will see what I can do.

File: ci/run_tests.sh [hunk 1]
```diff
@@ -46,6 +46,11 @@ do
     fi
 
     PYTEST_CMD="pytest -m \"$TYPE_PATTERN$PATTERN\" -n $NUM_JOBS -s --strict --durations=10 --junitxml=test-data-$TYPE.xml $TEST_ARGS $COVERAGE pandas"
+    if [[ "$(uname)" == "Linux" ]]; then
+        # This is required for the clipboard to work (X virtual frambuffer implements the X server API, required by the clipboard tools, without running an actual X server)
+        DISPLAY=:99.0
```
Comment (jreback) [inline ci/run_tests.sh]: so u prob need to start xvfb much earlier and background it

Comment (datapythonista) [reply ci/run_tests.sh]: I'm using xvfb-run, which starts it for the command, and stops it when the command finishes. It's what xvfb's got in the azure servers. I guess something else can be installed, but I'll see what I can do.

May be time to try again to call the tests just once (not twice for single and multi). :)

Comment (jbrockmendel) [reply ci/run_tests.sh]: I think I’ve seen similar things for matplotlib headless tests.  IIRC they add a 1-second sleep after this for the thing to start

**Comment by jorisvandenbossche** [comment]
Maybe try in a step before?

On Travis, they explicitly say in their docs it needs to be run in the "before_script" section (so the section where the tests are run)

**Comment by datapythonista** [comment]
I think the `sleep 1` will fix it, actually I did another run and worked correctly. Will play a bit more with it.

@jorisvandenbossche not an expert, but what I'm doing now is a bit different than what we've got in travis. In travis we just start the xvfb, and leaving it running. Here I use `xvfb-run`, which is like a Python context manager, it wraps a command with the server. If we use this, can't really move to another step, as I'm calling xvfb, and xvfb is calling the tests.

**Comment by datapythonista** [comment]
Confirmed that the problem is that `xvfb` fails to start because it's still running from the previous execution.

I can do a sleep for now if you want to get this in faster, but I think the right solution is the approach in #24106

I see that in travis the same solution doesn't seem to work, and clipboard tests are skipped. :\

**Comment by jorisvandenbossche** [comment]
> I see that in travis the same solution doesn't seem to work, and clipboard tests are skipped. :\

In the previous PR where I fixed those tests, I moved it back from run_tests.sh to before_script exactly because it was not working there (as said, it's in the documentation of travis it needs to be there). 

You're free to try to find a solution that works for both, but personally I wouldn't spend too much time on that if you have something that works for Azure.

**Comment by datapythonista** [comment]
Going for the simple hacking solution for now, so the tests start running.

If I have time I'll give another try to call pytest just once, that will solve the problem for azure, and simplify the CI significantly.

**Comment by datapythonista** [comment]
Closing this, as I think it make more sense to implement those changes in a fresh PR after #26949 is merged (which will simplify this significantly).

**Pull request closed without merging**